base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -1867,6 +1875,35 @@ fn rewind_to_height<T: BlockchainBackend>(db: &mut T, height: u64) -> Result<Vec
     Ok(removed_blocks)
 }
 
+fn reset_blockchain_databases<T: BlockchainBackend>(
```

Comment:
I would name this create_genesis_block / state and call this in `BlockchainDatabase::new` (line 241)

Then in reset_blockchain_database (line 1117) I would call reset_blockchain_databases and call create_genesis_block after.